solutions/cses-1736.mdx

```diff
@@ -0,0 +1,158 @@
+---
+id: cses-1736
+source: CSES
+title: Polynomial Queries
+author: Albert Ye
+---
+
+## Explanation
+
+Despite the strange operations we can still use a segment tree with lazy propagation
+to solve this problem.
+
+The range queries are vanilla lazy propagation, so we should focus more on the updates.
+
+We can split the "update" operation into two operations. One would be a basic range addition,
+and the other would be an incremental addition. Therefore, for each index $a_i$, we will need
+to keep track of two points: the total offset $s_i$ given only the offset sum (without increments),
+and the total increment $t_i$ which contains the sum of all increments from the incremental steps.
+
+When we propagate, the values of $s_i$ and $t_i$ for
+a node $k$ representing segment $[i, j]$ 
+when updating segment $[l, r]$ can be found by calculating $i-l$, as follows.
```

Comment:
segment = node in segtree
"updating segment $[l, r]$" refers to the update, so call it something else (maybe "subarray" or "range")

Reply:
sure thing